incorporating-accessibility-into-your-dev-process/focus-tests/solution/src/App.tsx

```diff
@@ -1,58 +1,117 @@
-import React, { FormEvent, useRef, useState } from "react";
+import React, { FormEvent, useId, useRef, useState } from "react";
+
+interface FormState {
+  firstName: string | undefined;
```

Comment:
No sure if this matters but i think its worth at least pointing out, there's a difference between `{firstName: string | undefined}` vs `{firstName?: string}`, `string | undefined` requires the key to be explicitly set to `undefined`. 

```ts
type A = {firstName: string | undefined}
type B = {firstName?: string}

const a: A = {firstName: undefined}
// errors
const aa: A = {}

const b: B = {firstName: undefined}
// doesn't error
const bb: B = {}
```

Maybe you want this, maybe you don't. just thought i'd bring it up incase there's instances below where you didn't want to have to explicitly say `undefined`

Reply:
Thanks for calling this out. I think I want what I have because I don’t want parts of the state to be updated—all of the state should be updated at once.